cloud/scope/cluster.go

```diff
@@ -102,10 +103,12 @@ func (s *ClusterScope) CreateVPC() (*vpcv1.VPC, error) {
 	options.SetName(s.IBMVPCCluster.Spec.VPC)
 	vpc, _, err := s.IBMVPCClients.VPCService.CreateVPC(options)
 	if err != nil {
+		record.Warnf(s.IBMVPCCluster, "FailedCreate", "Failed vpc creation - %v", err)
 		return nil, err
 	} else if err := s.updateDefaultSG(*vpc.DefaultSecurityGroup.ID); err != nil {
 		return nil, err
```

Comment:
Need Warnf for this failure as well

Reply:
1. Added `Warnf` for above failure.
2. Updated record reasons to be more specific.

How about we merge this version of the code and add fake recorder + enhance test cases as part of another PR?

Reply:
better to write test cases with the code itself, for this case we can add later